scrapy/extensions/corestats.py

```diff
@@ -21,10 +22,16 @@ def from_crawler(cls, crawler):
         return o
 
     def spider_opened(self, spider):
-        self.stats.set_value('start_time', datetime.datetime.utcnow(), spider=spider)
+        start_time = datetime.datetime.now()
+        self.start_time = start_time
+        self.stats.set_value('start_time', start_time, spider=spider)
 
     def spider_closed(self, spider, reason):
-        self.stats.set_value('finish_time', datetime.datetime.utcnow(), spider=spider)
+        finish_time = datetime.datetime.now()
+        elapsed_time = finish_time - self.start_time
```

Comment:
Maybe fetching ``start_time`` from stats would be more robust? That would allow to remove start_time and finish_time variables.